hazelcast/src/main/java/com/hazelcast/cache/impl/nearcache/impl/store/NearCacheDataRecordStore.java

```diff
@@ -79,6 +81,10 @@ protected NearCacheDataRecord valueToRecord(V value) {
 
     @Override
     protected V recordToValue(NearCacheDataRecord record) {
+        if (record.getValue() == null) {
+            nearCacheStats.incrementMisses();
+            return (V) NULL_OBJECT;
```

Comment:
What about handling `null` <=> `NULL_OBJECT` in `NearCache` level not `NearCacheRecordStore` level?

I mean, `NearCacheRecordStore` should not allow any `null` but `NearCache` handles convertion between  `null` and `NULL_OBJECT`. 

* If the value on the put is `null`, `NearCache` converts value to `NULL_OBJECT` and passes it to underlying `NearCacheRecordStore`. 
* If the value is `NULL_OBJECT` on get returning from underlying `NearCacheRecordStore`, `NearCache` returns `null` to caller.